src/scripts/settings/webSettings.js

```diff
@@ -1,27 +1,11 @@
-define(['appStorage', 'events'], function (appStorage, events) {
-    'use strict';
-
-    var data;
-
-    function getConfig() {
-        if (data) {
-            return data;
-        }
-
-        fetch("/config.json").then(function (response) {
-            data = response.json();
-        })
-
-        return data;
-    }
-
-    function WebSettings() {
-        getConfig();
-    }
-
-    WebSettings.prototype.enableMultiServer = function () {
-        return getConfig().multiServer || false;
-    };
-
-    return new WebSettings();
-});
+function getConfig() {
+    return fetch("/config.json?nocache=" + new Date().getUTCMilliseconds()).then(function (response) {
```

Comment:
Any reason for not storing the value of response to a local variable?
IMHO it should request settings once and use them until a page is completely refreshed.

Reply:
> The intent is to apply config changes immediately, which won't work if any kind of browser cache is used during the fetch.

Reply:
I see the intent, but it would make running the UI slower by adding one more request.
The case when you update the settings is usually rare, you don't go changing them all the time.

So refreshing a page manually when settings are changed isn't a big deal, but adding more lag is IMHO.

Reply:
Refreshing the page manually will always hit the browser cache with the same URL and we can't ask every user to update their browser cache whenever they change these settings.

Reply:
> You could simply return Promise.resolve(data) if data is fetched.

```js
let data;

function getConfig() {
    if (data) {
        return Promise.resolve(data);
    }

    return fetch("/config.json?nocache=" + new Date().getUTCMilliseconds()).then(function (response) {
        data = response.json();
        return data;
    });
}
...
```

Reply:
That would be a nice middle ground, but it depends on if we want to refresh the settings during a user session or not. This wouldn't refresh until they fully reloaded the website.

Reply:
I went with this method, I think once per session should be enough now that you mention it.